stdin-wrapper

```diff
@@ -1,20 +1,28 @@
 #!/usr/bin/env bash
 
-# Author: w0rp <devw0rp@gmail.com>
+# Authors: w0rp <devw0rp@gmail.com>, hauleth <lukasz@niemier.pl>
 # Description: This script implements a wrapper for any program which does not accept
 #   stdin input on most Unix machines. The input to the script is read to a
 #   temporary file, and the first argument sets a particular file extension
 #   for the temporary file.
 
+set -eu
+
 # All of the following arguments are read as command to run.
 file_extension="$1"
 shift
 
-temp_file=$(mktemp --tmpdir "ale-XXXXXXXXX$file_extension")
-trap 'rm $temp_file' EXIT
+temp_dir=$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')
```

Comment:
@w0rp I think that instead of using `javac` as a linter we should add support for Sonar or other external static analyzers. `javac` does poor job as a linter.